frontend/src/libs/components/radio/radio.tsx

```diff
@@ -0,0 +1,70 @@
+import {
+  type Control,
+  type FieldErrors,
+  type FieldPath,
+  type FieldValues,
+} from 'react-hook-form';
+
+import { getValidClassNames } from '~/libs/helpers/helpers.js';
+import { useFormController } from '~/libs/hooks/hooks.js';
+
+import common from './common.module.scss';
+import radio from './radio.module.scss';
+
+const styles = { ...common, ...radio };
+
+type Properties<T extends FieldValues, N extends FieldPath<T>> = {
+  control: Control<T, null>;
+  errors?: FieldErrors<T>;
+  label?: string;
+  name: N;
+  beDisabled?: boolean;
+  value?: T[N];
+};
+
+const Radio = <T extends FieldValues, N extends FieldPath<T>>({
+  control,
+  errors,
+  label,
+  name,
+  value,
+  beDisabled = false,
+}: Properties<T, N>): JSX.Element => {
+  const { field } = useFormController({ name, control });
+
+  const error = errors?.[name]?.message;
+  const hasLabel = Boolean(label);
+  const hasError = Boolean(error);
+  const isChecked = field.value === value;
+
+  return (
+    <span className={styles.container}>
+      <label
+        className={getValidClassNames(styles.label, beDisabled && 'disabled')}
+      >
+        <input
+          className={getValidClassNames(styles.input, styles.radio)}
+          {...field}
+          type="radio"
+          value={value}
+          disabled={beDisabled || undefined}
+          checked={isChecked}
+          aria-disabled={beDisabled || undefined}
```

Comment:
```suggestion
          disabled={isDisabled}
          checked={isChecked}
          aria-disabled={isDisabled}
```
I believe that we don't need undefined here, isDisabled has default value

Reply:
without undefined we would have aria-disabled="false" , disabled="false" in our dom. By having undefined I make sure that dom will remove these properties, if beDisabled (isDisabled) is false.

Reply:
![изображение](https://github.com/BinaryStudioAcademy/bsa-2023-towhub/assets/81959259/e2bd799f-2d31-432a-aa36-d4592529ec8d)